### PR TITLE
In-memory engine: add metrics for prepare for write duration

### PR DESCRIPTION
**Pull request by SpadeA-Tang**
### What is changed and how it works?

Issue Number: Ref #16141

What's Changed:

```commit-message
add metrics for prepare for write duration
```
image

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by overvenus — APPROVED** [review]

File: components/region_cache_memory_engine/src/metrics.rs [hunk 1]
```diff
@@ -69,6 +69,12 @@ lazy_static! {
         exponential_buckets(0.00001, 2.0, 20).unwrap()
     )
     .unwrap();
+    pub static ref PREPARE_FOR_WRITE_DURATION_HISTOGRAM: Histogram = register_histogram!(
```
Comment (LykxSassinator) [inline components/region_cache_memory_engine/src/metrics.rs:72]:
```suggestion
    pub static ref RANGE_PREPARE_FOR_WRITE_DURATION_HISTOGRAM: Histogram = register_histogram!(
```

**Review by LykxSassinator — COMMENTED** [review]
Rest LGTM

**Review by LykxSassinator — APPROVED** [review]

**Pull request merged**